src/jsonld/event.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC } from 'react';
+import Head from 'next/head';
+
+import markup from '../utils/markup';
+import formatIfArray from '../utils/formatIfArray';
+import { Address } from '../types';
+import buildAddress from '../utils/buildAddress';
+
+type Location = {
+  name: string;
+  address: Address;
+  sameAs?: string;
+};
+
+export interface EventJsonLdProps {
+  name: string;
+  startDate: string;
+  endDate: string;
+  location: Location;
+  url?: string;
+  description?: string;
+  images?: string[];
+}
+
+const buildLocation = (location: Location) => `
+  "location": {
+    "@type": "Place",
+    "name": "${location.name}",
+    ${buildAddress(location.address)}
+    ${location.sameAs ? `"sameAs": "${location.sameAs}"` : ``}
+  },
+`;
+
+const EventJsonLd: FC<EventJsonLdProps> = ({
+  name,
+  startDate,
+  endDate,
+  location,
+  url,
+  description,
+  images,
+}) => {
+  const jslonld = `{
```

Comment:
I think we will run into another issue here we trailing commas 😅 

If we move things around we should be able to ensure it will be ok even when all optional fields are not set. Something like this should work:
```
"@context": "http://schema.org",
    "@type": "Event",
    "startDate": "${startDate}",
    "endDate": "${endDate}",
    ${buildLocation(location)}
    ${images ? `"image":${formatIfArray(images)},` : ``}
    ${url ? `"url": "${url}",` : ``}
    ${description ? `"description": "${description}",` : ``}
    "name": "${name}"
```